catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -110,14 +117,40 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
 
         SolrQuery query = getSolrQuery(request, filterDelegateFactory.newInstance(resolver));
 
+        boolean isFacetedQuery = false;
+        Serializable textFacetPropRaw = request.getPropertyValue(EXPERIMENTAL_FACET_FIELDS_KEY);
+
+        if(textFacetPropRaw != null && textFacetPropRaw instanceof FacetProperties) {
+            FacetProperties textFacetProp = (FacetProperties) textFacetPropRaw;
+            isFacetedQuery = true;
+
+            textFacetProp.getFacetFields().stream()
+                    .map(facet -> !facet.endsWith("_txt") ? facet + "_txt" : facet)
```

Comment:
It already is a constant.

https://github.com/codice/ddf/blob/master/catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SchemaFields.java#L44